pytket/extensions/cutensornet/tnstate/simulation.py

```diff
@@ -223,6 +223,68 @@ def _get_qubit_partition(
     return qubit_partition
 
 
+def _apply_kahypar_bisection(
+    graph: nx.Graph,
+) -> tuple[list[Qubit], list[Qubit]]:
+    """Use KaHyPar to obtain a bisection of the graph.
+
+    Returns:
+        Two lists, each containing the vertices in either group of the bisection.
+    """
+    vertices = list(graph.nodes)
+    edges = list(graph.edges)
+    weight_dict = nx.get_edge_attributes(graph, "weight")
+    qubit_dict = {q: i for i, q in enumerate(vertices)}
+
+    num_vertices = len(vertices)
+    num_edges = len(edges)
+    k = 2  # Number of groups in the partition
+    epsilon = 0.03  # Imbalance tolerance
+
+    # Special case where the graph has no edges; KaHyPar cannot deal with it
+    if num_edges == 0:
+        # Just split the list of vertices in half
+        return (vertices[: num_vertices // 2], vertices[num_vertices // 2 :])
+
+    # KaHyPar expects the list of edges to be provided as a continuous set of vertices
+    # ``edge_stream`` where ``edge_indices`` indicates where each new edge begins
+    # (the latter is necessary because KaHyPar can accept hyperedges)
+    edge_stream = [qubit_dict[vertex] for edge in edges for vertex in edge]
+    edge_indices = [0] + [2 * (i + 1) for i in range(num_edges)]
+    edge_weights = [weight_dict[edge] for edge in edges]
+    vertex_weights = [1 for _ in range(num_vertices)]
+
+    hypergraph = kahypar.Hypergraph(

```

Comment:
It says here "hypergraph", but it's just a graph (KaHypar can deal with hypergraphs). Happy to change the variable's name to "graph" if this seems confusing (although its still an instance of a Hypergraph, as indicated by its constructor method).

Reply:
It looks fine to me!